test/sambah/SAMBAH_Regression.ipynb

```diff
@@ -0,0 +1,363 @@
+    "\n",
+    "* Variable subsetting, including required variables.\n",
+    "* Temporal subsetting.\n",
+    "* Bounding box spatial subsetting.\n",
+    "\n",
+    "## Prerequisites\n",
+    "\n",
+    "The dependencies for this notebook are listed in the [environment.yaml](./environment.yaml). To test or install locally, create the papermill environment used in the automated regression testing suite:\n",
+    "\n",
+    "`conda env create -f ./environment.yaml && conda activate papermill-sambah`\n",
+    "\n",
+    "A `.netrc` file must also be located in the `test` directory of this repository."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "009f1df8",
+   "metadata": {},
+   "source": [
+    "## Import requirements:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7ce7f640",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-05-17T17:52:02.432376Z",
+     "start_time": "2024-05-17T17:52:01.930482Z"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from datetime import datetime\n",
+    "from os.path import exists\n",
+    "\n",
+    "from harmony import BBox, Client, Collection, Dimension, Environment, Request\n",
+    "\n",
+    "from utilities import (\n",
+    "    compare_results_to_reference_file,\n",
+    "    remove_results_files,\n",
+    "    submit_and_download,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "00758d6f",
+   "metadata": {},
+   "source": [
+    "## Set default parameters:\n",
+    "\n",
+    "`papermill` requires default values for parameters used on the workflow. In this case, `harmony_host_url`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "45f380c6",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-05-17T17:53:16.067063Z",
+     "start_time": "2024-05-17T17:53:16.061847Z"
+    },
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "harmony_host_url = 'https://harmony.uat.earthdata.nasa.gov'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9237f449",
+   "metadata": {},
+   "source": [
+    "### Identify Harmony environment (for easier reference):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4420bb6e",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-05-17T17:53:23.137293Z",
+     "start_time": "2024-05-17T17:53:20.917984Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "host_environment = {\n",
+    "    'http://localhost:3000': Environment.LOCAL,\n",
+    "    'https://harmony.uat.earthdata.nasa.gov': Environment.UAT,\n",
+    "    'https://harmony.earthdata.nasa.gov': Environment.PROD,\n",
+    "}\n",
+    "\n",
+    "data_environment = 'PROD' if harmony_host_url == '' else 'UAT'\n",
```

Comment:
From this the notebook is going to always use the UAT test data values because of this condition, but that feels a bit wonky. That means that a run in production will trigger errors because it will be trying to find UAT data. Is this because there aren't equivalent production data available, yet?

Reply:
fixed the environment mutability, mimicking other test notebooks.